common/djangoapps/util/enterprise_helpers.py

```diff
@@ -203,6 +203,12 @@ def inner(request, course_id, *args, **kwargs):
         # Redirect to the consent URL, if consent is required.
         consent_url = get_enterprise_consent_url(request, course_id)
         if consent_url:
+            user = request.user
+            LOGGER.warning(
+                u'User %s cannot access the course %s because they have not granted consent',
+                user.get('real_user', user),
+                course_id,
+            )
             return redirect(consent_url)
```

Comment:
Reviewers: this decorator always redirects to the same URL after the user grants consent, while some of the custom implementations redirect back to the originating page. Should this method be changed to always go back to the URL that the user originally requested?